contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,251 @@
+     * @param _metadata_uri ERC1155 metadata uri
+     */
+    constructor(
+        ERC20Votes _token,
+        string memory _metadata_uri
+    ) ERC1155(_metadata_uri) {
+        token = _token;
+    }
+
+    /**
+     * @dev Deposits and delegates voting power to the multiple delegatees.
+     * @param delegateeAmounts The list of delegatee addresses and corresponding list of ERC20 voting power amount amounts to delegate.
+     */
+    function depositMulti(
+        DelegateeAmount[] calldata delegateeAmounts
+    ) external {
+        uint256 delegateesLength = delegateeAmounts.length;
+
+        require(
+            delegateesLength > 0,
+            "DepositMulti: You should pick at least one delegatee"
+        );
+
+        uint256[] memory ids = new uint256[](delegateesLength);
+        uint256[] memory amounts = new uint256[](delegateesLength);
+
+        for (uint256 index = 0; index < delegateesLength; index++) {
+            address delegatee = delegateeAmounts[index].delegatee;
+            uint256 amount = delegateeAmounts[index].amount;
+
+            createProxyDelegatorAndTransfer(delegatee, amount);
+
+            ids[index] = uint256(uint160(delegatee));
+            amounts[index] = amount;
+        }
+
+        mintBatch(msg.sender, ids, amounts);
+    }
+
+    /**
+    * @dev Re-delegates voting power between delegatees.
+    * @param delegateePairs The list of current delegatee addresses (from which the voting power is withdrawn) 
+    * and target delegatee addresses (which the voting power will be delegated).
+    */
+    function reDeposit(SourceTargetDelegatee[] calldata delegateePairs) external {
+        uint256 delegateePairsLength = delegateePairs.length;
+
+        require(
+            delegateePairsLength > 0,
+            "ReDeposit: You should pick at least one source and target delegatee pair"
+        );
+
+        uint256[] memory sourceIds = new uint256[](delegateePairsLength);
+        uint256[] memory targetIds = new uint256[](delegateePairsLength);
+        uint256[] memory amounts = new uint256[](delegateePairsLength);
+
+        for (uint index = 0; index < delegateePairsLength; index++) {
+            address from = delegateePairs[index].source;
+            address to = delegateePairs[index].target;
+
+            uint256 amount = getBalanceForDelegatee(from);
+            sourceIds[index] = uint256(uint160(from));
+            targetIds[index] = uint256(uint160(to));
+            amounts[index] = amount;
+
+            transferBetweenDelegators(from, to, amount);
+
+            deployProxyDelegatorIfNeeded(to);
+        }
+
+        burnBatch(msg.sender, sourceIds, amounts);
+        mintBatch(msg.sender, targetIds, amounts);
+    }
+
+    /**
+     * @dev Withdraws delegated ERC20 voting power from proxy delegators to the actual delegator
+     * @param delegatees List of delegatee addresses
+     */
+    function withdrawMulti(address[] calldata delegatees) external {
```

Comment:
This should probably allow partial withdraws by taking an array of amounts as well.

Reply:
updated: 

https://github.com/ensdomains/governance/blob/4e4c7cbc89d82aa9f035a7abcc7a1854432eb423/contracts/ERC20MultiDelegate.sol#L29-L32

https://github.com/ensdomains/governance/blob/4e4c7cbc89d82aa9f035a7abcc7a1854432eb423/contracts/ERC20MultiDelegate.sol#L290